src/ophys_etl/modules/neuropil_correction/__main__.py

```diff
@@ -0,0 +1,227 @@
+            logging.info(
+                "r=%f err=%f it=%d",
+                results["r"],
+                results["err"],
+                results["it"],
+            )
+
+            r = results["r"]
+            fc = roi - r * neuropil
+            RMSE_array[n] = results["err"]
+            r_vals[n] = results["r_vals"]
+
+            if r > 1:
+                logging.info(f"Estimated r value > 1, r = {r}")
+            debug_plot(
+                os.path.join(plot_dir, "initial_%04d.png" % n),
+                roi,
+                neuropil,
+                fc,
+                r,
+                results["r_vals"],
+                results["err_vals"],
+            )
+
+            # mean of the corrected trace must be positive
+            if fc.mean() > 0:
+                r_array[n] = r
+                corrected[n, :] = fc
+            else:
+                logging.warning(
+                    "fc has negative baseline, skipping this r value"
+                )
+
+        # fill in empty r values
+        for n in range(num_traces):
+            roi = roi_traces["data"][n]
+            neuropil = neuropil_traces["data"][n]
+
+            if r_array[n] is np.nan:
+                logging.warning(
+                    "fc had negative baseline %d. Setting r to zero.", n
+                )
+                r_array[n] = 0
+                corrected[n, :] = roi
+
+            # save a debug plot
+            debug_plot(
+                os.path.join(plot_dir, "final_%04d.png" % n),
+                roi,
+                neuropil,
+                corrected[n, :],
+                r_array[n],
+            )
+
+            # one last sanity check
+            eps = -0.0001
+            if np.mean(corrected[n, :]) < eps:
+                raise Exception(
+                    "Trace %d baseline is still negative value after"
+                    "correction" % n
+                )
+
+            if r_array[n] < 0.0:
+                raise Exception("Trace %d ended with negative r" % n)
+
+        r_array = r_array.astype(float)
+
+        # flag cells with unconverged r values (r>1) and fill in with
+        # mean r value across all other cells of the same experiment.
+        # recalculate neuropil_corrected trace and RMSE
+        if any(r_array > 1):
+            logging.info(
+                f"Number of unconverged r values > 1: {sum(r_array > 1)}"
+                "Filling in unconverged r values with mean r value"
+                "Recalculating corrected trace and RMSE"
+            )
+            corrected, r_array, RMSE_array = fill_unconverged_r(
+                corrected,
+                roi_traces["data"][()],
+                neuropil_traces["data"][()],
+                r_array,
+            )
+
+        ######################################################################
+        # write out processed data
+        try:
+            self.args["neuropil_correction"] = os.path.join(
```

Comment:
this is weird. It shouldn't be added to `self.args` since it's not part of the schema. Can just set to a variable?